pkg/providers/cloudstack/validator_test.go

```diff
@@ -0,0 +1,90 @@
+package cloudstack
+
+import (
+	"context"
+	_ "embed"
+	"path"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/providers/cloudstack/mocks"
+)
+
+const (
+	testClusterConfigMainFilename = "cluster_main.yaml"
+	testDataDir                   = "testdata"
+)
+
+var testZone = v1alpha1.CloudStackResourceRef{
+	Type:  "name",
+	Value: "zone1",
+}
+
+func givenCloudMonkeyMock(t *testing.T) *mocks.MockProviderCmkClient {
+	ctrl := gomock.NewController(t)
+	return mocks.NewMockProviderCmkClient(ctrl)
+}
+
+func givenDatacenterConfig(t *testing.T, fileName string) *v1alpha1.CloudStackDatacenterConfig {
+	deploymentConfig, err := v1alpha1.GetCloudStackDatacenterConfig(path.Join(testDataDir, fileName))
```

Comment:
I would suggest avoiding this and declaring the datacenter and machine configs structs in code directly

Reply:
Sounds good